snapcraft/cli/_channel_map.py

```diff
@@ -0,0 +1,267 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from collections import OrderedDict
+from typing import List, NamedTuple, Optional
+
+from tabulate import tabulate
+
+from snapcraft.storeapi.v2.channel_map import (
+    ChannelMap,
+    MappedChannel,
+    Revision,
+    SnapChannel,
+)
+
+
+_HINTS_T = NamedTuple(
```

Comment:
I'd suggest using mypy to for constant checks with `Final`.

Either something like `_SYMBOL_CLOSED: Final[str] = "-"`
Or If you prefer, bundle them into a class:
```
from typing import Final
class Hints:
  CLOSED: Final[str] = "-"
  FOLLOWING: Final[str] = "↑"
...
```

Reply:
Sure.